src/main/java/org/opensearch/security/auth/BackendRegistry.java

```diff
@@ -114,6 +124,14 @@ public void onRemoval(RemovalNotification<AuthCredentials, User> notification) {
             })
             .build();
 
+        if (!existingUserCache.isEmpty()) {
```

Comment:
Nit: As `userCache.putAll(existingUserCache)` is a no-op for an empty map argument, it would be safe to remove the if condition.

This would even improve test coverage, as we have fewer code branches to cover in our code base.